src/api/exec.c

```diff
@@ -1,9 +1,46 @@
 #include "minishell.h"
-// #include <unistd.h>
-// #include <stdlib.h>
-
-// int main(int ac, char *av[], char *ep[])
-// {
-// 	execve("/bin/ls", (char *[]){"-al", "/Users/youkim", NULL}, ep);
-// 	return (0);
-// }
\ No newline at end of file
+
+
+static void	api_exec_cmd_inner(t_AST_COMMAND *cmd, t_dict *env)
+{
+	int		i;
+	char	**envp;
+	char	**names;
+	(void)cmd;
+
+	names = new_path_with_name(env, "ls"); //cmd->name.text);
+	envp = new_arr_env(env);
+	i = -1;
+	while (names[++i])
+		execve(names[i], (char *[]){"~", NULL}, envp);
+	del_arr(envp);
+}
+
+bool	is_parent(pid_t pid)
+{
+	return (pid > 0);
+}
+
+void	api_exec_cmd(t_AST_COMMAND *cmd, t_dict *env)
+{
+	pid_t	pid;
+	(void)cmd; (void)env;
+
+	pid = fork();
+	if (is_parent(pid))
+	{
+		printf("I'm parent and waiting for child\n");
+		// close(pipe[PIPE_WRITE]);
+		// dup2(pipe[PIPE_READ], STDIN_FILENO);
+		waitpid(pid, NULL, WNOHANG);
+		printf("yo child's dead\n");
```

Comment:
```suggestion
		wait_flag = waitpid(pid, NULL, WNOHANG);
		if (wait_flag == pid)
		    printf("yo child's dead\n");
		// 0을 반환하면 아직 자식프로세스는 돌고 있다는 의미로, 다른 처리 필요
```
아직 진행중이라고 하셨지만... 이거 waitpid에 WNOHANG 쓰면 무조건 리턴이라서
0인지 pid를 받았는지 확인해야 제대로 사용했다고 볼 수 있어요!
지금 사용하신건 자식 프로세스가 돌고 있어도 아래 printf를 실행합니닷

만약 정말 기다리기를 원하시면 WNOHANG 옵션을 빼는 게 좋을 것 같아요.

Reply:
이것도 전과제에서 무지성으로 복붙한거네요 ㄷㄷ 다시 수정해봐야겠습니다